api/src/main/java/io/keploy/service/GrpcService.java

```diff
@@ -155,6 +164,107 @@ public static void put(Service.TestCaseReq testCaseReq) {
+     *
+     * @param testCaseReq the test case req object which contains all the information about the test case
+     * @return the boolean value which indicates if the test case should be recorded or discarded
+     */
+    public static boolean isValidTestCaseToBeRecorded(Service.TestCaseReq testCaseReq) {
+        // acc    rej        record/discard
+        // true   true   =>  discard
+        // true   false  =>  record
+        // false  true   =>  discard
+        // false  false  =>  discard
+        return doesFollowAcceptanceRegex(testCaseReq) && !doesFollowRejectionRegex(testCaseReq);
+
+    }
+
+    /**
+     * Does follow acceptance regex boolean.
+     * This method is used to check if the test case url and header matches the acceptance regex.
+     *
+     * @param testCaseReq the test case req object which contains all the information about the test case
+     * @return the boolean value which indicates if the test case url and header matches the acceptance regex
+     */
+    public static boolean doesFollowAcceptanceRegex(Service.TestCaseReq testCaseReq) {
+        boolean isIncluded = false;
+        Filter filter = k.getCfg().getApp().getFilter();
+
+        // get test case url regex
+        if (filter.getAcceptUrlRegex() != null && filter.getAcceptUrlRegex().length > 0) {
+            // check if test case url match the regex
+            String testCaseUrl = testCaseReq.getHttpReq().getURL();
+            for (String value : filter.getAcceptUrlRegex()) {
+                Pattern pattern = Pattern.compile(value);
+                if (pattern.matcher(testCaseUrl).matches()) {
+                    isIncluded = true;
+                    break;
+                }
+            }
+        }
+
+        // get test case header regex
+        if( !isIncluded && filter.getAcceptHeaderRegex() != null && filter.getAcceptHeaderRegex().length > 0) {
+            // check if test case header match the regex
+            Map<String, Service.StrArr> headerMap = testCaseReq.getHttpReq().getHeaderMap();
+            for (Map.Entry<String, Service.StrArr> entry : headerMap.entrySet()) {
+                String key = entry.getKey();
+                for (String value : filter.getAcceptHeaderRegex()) {
+                    Pattern pattern = Pattern.compile(value);
+                    if (pattern.matcher(key).matches()) {
+                        isIncluded = true;
+                        break;
+                    }
+                }
+            }
+        }
+        return isIncluded;
+    }
+
+    /**
+     * Does follow rejection regex boolean.
+     * This method is used to check if the test case url and header matches the rejection regex.
+     *
+     * @param testCaseReq the test case req object which contains all the information about the test case.
+     * @return the boolean value which indicates if the test case url and header matches the rejection regex
+     */
```

Comment:
Here, `doesFollowRejectionRegex` <- it is returning false if any of the two `urlRegex` or `headerRegex` matches. But it should check for both conditions.